hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/DefineGroupsLayout.java

```diff
@@ -331,22 +334,29 @@ private void setGroupsValidation(final RolloutGroupsValidation validation) {
         }
 
         // validate the single groups
-        final int maxTargets = quotaManagement.getMaxTargetsPerRolloutGroup();
         final boolean hasRemainingTargetsError = validationStatus == ValidationStatus.INVALID;
+        singleGroupsValidation(validation, lastRow, hasRemainingTargetsError);
+    }
+
+    private void singleGroupsValidation(final RolloutGroupsValidation validation, final GroupRow lastRow,
+            final boolean remainingError) {
+        final int maxTargets = quotaManagement.getMaxTargetsPerRolloutGroup();
+
         for (int i = 0; i < groupRows.size(); ++i) {
             final GroupRow row = groupRows.get(i);
             // do not mask the 'remaining targets' error
-            if (hasRemainingTargetsError && row.equals(lastRow)) {
+            if (remainingError && row.equals(lastRow)) {
                 continue;
             }
             row.resetError();
-            final Long count = groupsValidation.getTargetsPerGroup().get(i);
-            if (count != null && count > maxTargets) {
-                row.setError(i18n.getMessage(MESSAGE_ROLLOUT_MAX_GROUP_SIZE_EXCEEDED, maxTargets));
-                setValidationStatus(ValidationStatus.INVALID);
+            if (validation != null) {
```

Comment:
You could use the _groupsValidation_ member field here, so you could remove the _validation_ method parameter